sql: default to SHARED service in CREATE TENANT

dt opened the pull request:
creating directly into a desired service mode avoids an extra step when setting up a new tenant, and opens up a path to creating directly in SHARED by default in the future once creations which do not wish for immediate service can opt into one of the other modes during creation.

Release note: none.
Epic: none.

dt committed:
sql/parser: define WITH options on CREATE TENANT

Release note: none.
Epic: none.

dt committed:
sql: plumb service option to CREATE TENANT

Release note: none.
Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/119636)
<!-- Reviewable:end -->

stevendanna reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

We could have a little more in the commit message about why we think this is useful, bit I think it clearly is.

dt committed:
sql: make SHARED default service for CREATE TENANT

Making SHARED the default saves steps when creating a new UA tenant, and
avoids needing to explain service modes to UA operators.

Operators who need to directly manage service modes and lifecycle can
elect to use the WITH SERVICE option to choose a different mode.

Release note (enterprise change): CREATE VIRTUAL CLUSTER now automatically creates
the cluster in SHARED SERVICE mode, instead of requiring subsequent START SERVICE.

Epic: none.

dt commented:
@stevendanna I added an additional commit with a larger behavior change that I think is the real motivation. RFAL.

stevendanna commented on pkg/ccl/streamingccl/streamingest/stream_ingestion_manager_test.go:
```diff
@@ -46,7 +46,7 @@ func TestRevertTenantToTimestamp(t *testing.T) {
 	systemSQL.Exec(t, "SET CLUSTER SETTING kv.closed_timestamp.side_transport_interval ='100ms'")
 	systemSQL.Exec(t, "SET CLUSTER SETTING kv.rangefeed.closed_timestamp_refresh_interval ='100ms'")
-	systemSQL.Exec(t, "CREATE VIRTUAL CLUSTER target")
+	systemSQL.Exec(t, "CREATE VIRTUAL CLUSTER target WITH SERVICE NONE")
```
Can we open an issue for the problem we discovered here? I think we found some pretty suspect behavior on rapid state changes.

dt replied:
https://github.com/cockroachdb/cockroach/issues/119724

stevendanna reviewed (approved):
I like the addition of this with option.

I think making SHARED the default makes the most sense. EXTERNAL certainly shouldn't be the default, and every other user will want SHARED.

As I believe you are already on top of we need to coordinate with @JeffSwenson about how this impacts his work.

dt commented:
TFTR!

bors r+

dt commented:
bors r-
